components/src/preact/mutationsOverTime/mutations-over-time.tsx

```diff
@@ -63,15 +63,21 @@ export const MutationsOverTimeInner: FunctionComponent<MutationsOverTimeProps> =
     const lapis = useContext(LapisUrlContext);
     const { lapisFilter, sequenceType, granularity, lapisDateField } = componentProps;
 
-    const { data, error, isLoading } = useWebWorker<MutationOverTimeQuery, MutationOverTimeWorkerResponse>(
-        {
+    const messageToWorker = useMemo(() => {
+        return {
             lapisFilter,
             sequenceType,
             granularity,
             lapisDateField,
             lapis,
-        },
-        new MutationOverTimeWorker() as Worker,
+        };
+    }, [granularity, lapis, lapisDateField, lapisFilter, sequenceType]);
+
+    const workerProvider = useCallback(() => new MutationOverTimeWorker() satisfies Worker, []);
+
+    const { data, error, isLoading } = useWebWorker<MutationOverTimeQuery, MutationOverTimeWorkerResponse>(
+        messageToWorker,
+        workerProvider,
     );
```

Comment:
I think we can do even better by reusing the same worker:

```suggestion
    const worker = useMemo(() => new MutationOverTimeWorker(), []);

    const { data, error, isLoading } = useWebWorker<MutationOverTimeQuery, MutationOverTimeWorkerResponse>(
        messageToWorker,
        worker,
    );
```

This requires splitting the `useEffect` in `useWebWorker` in two parts:
```
    useEffect(() => {
        worker.onmessage = ...
        worker.onmessageerror = ...

        // no return!
    }, [worker]);

    useEffect(() => {
        worker.postMessage(messageToWorker);
    }, [messageToWorker, worker]);
```